python/tests/test_table_read.py

```diff
@@ -17,6 +17,50 @@ def test_read_simple_table_by_version_to_dict():
     assert dt.to_pyarrow_dataset().to_table().to_pydict() == {"value": [1, 2, 3]}
 
 
+def test_get_files_partitioned_table():
+    table_path = "../rust/tests/data/delta-0.8.0-partitioned"
+    dt = DeltaTable(table_path)
+    partition_filters = [("day", "=", "3")]
+    assert dt.files_by_partitions(partition_filters=partition_filters) == [
+        "year=2020/month=2/day=3/part-00000-94d16827-f2fd-42cd-a060-f67ccc63ced9.c000.snappy.parquet"
+    ]
+    partition_filters = [("day", "!=", "3")]
+    assert dt.files_by_partitions(partition_filters=partition_filters) == [
+        "year=2020/month=1/day=1/part-00000-8eafa330-3be9-4a39-ad78-fd13c2027c7e.c000.snappy.parquet",
+        "year=2020/month=2/day=5/part-00000-89cdd4c8-2af7-4add-8ea3-3990b2f027b5.c000.snappy.parquet",
+        "year=2021/month=12/day=20/part-00000-9275fdf4-3961-4184-baa0-1c8a2bb98104.c000.snappy.parquet",
+        "year=2021/month=12/day=4/part-00000-6dc763c0-3e8b-4d52-b19e-1f92af3fbb25.c000.snappy.parquet",
+        "year=2021/month=4/day=5/part-00000-c5856301-3439-4032-a6fc-22b7bc92bebb.c000.snappy.parquet",
+    ]
+    partition_filters = [("day", "in", ["3", "20"])]
+    assert dt.files_by_partitions(partition_filters=partition_filters) == [
+        "year=2020/month=2/day=3/part-00000-94d16827-f2fd-42cd-a060-f67ccc63ced9.c000.snappy.parquet",
+        "year=2021/month=12/day=20/part-00000-9275fdf4-3961-4184-baa0-1c8a2bb98104.c000.snappy.parquet",
+    ]
+    partition_filters = [("day", "not in", ["3", "20"])]
+    assert dt.files_by_partitions(partition_filters=partition_filters) == [
+        "year=2020/month=1/day=1/part-00000-8eafa330-3be9-4a39-ad78-fd13c2027c7e.c000.snappy.parquet",
+        "year=2020/month=2/day=5/part-00000-89cdd4c8-2af7-4add-8ea3-3990b2f027b5.c000.snappy.parquet",
+        "year=2021/month=12/day=4/part-00000-6dc763c0-3e8b-4d52-b19e-1f92af3fbb25.c000.snappy.parquet",
+        "year=2021/month=4/day=5/part-00000-c5856301-3439-4032-a6fc-22b7bc92bebb.c000.snappy.parquet",
+    ]
+    partition_filters = [("day", "not in", ["3", "20"]), ("year", "=", "2021")]
+    assert dt.files_by_partitions(partition_filters=partition_filters) == [
+        "year=2021/month=12/day=4/part-00000-6dc763c0-3e8b-4d52-b19e-1f92af3fbb25.c000.snappy.parquet",
+        "year=2021/month=4/day=5/part-00000-c5856301-3439-4032-a6fc-22b7bc92bebb.c000.snappy.parquet",
+    ]
+    partition_filters = [("invalid_operation", "=>", "3")]
+    with pytest.raises(Exception):
```

Comment:
I could just be misunderstanding some of the Python code here, but would it be possible for the binding to raise a typed exception that indicates that the filter contains an invalid operand?

If this is not how other underlying errors from the Rust layer are mapped into the Python binding, please ignore my noise :smile:

Reply:
We need to provide more information on the exceptions raised by Rust in the Python tests 👍